### PR TITLE
feature: get contacts

### PR DESCRIPTION
**Pull request by minh-rakuten**
# Description
Allow miniapp to get contact list from hostapp.
Can be tested with JS Sample app version 1.6.

## Links
MINI-2296

# Checklist
- [x] I have read the [contributing guidelines](CONTRIBUTING.md).
- [x] I have completed the [SDK Development Learning Path](CONTRIBUTING.md#SDK-Development-Learning-Path) (if submitting a feature PR)
- [x] I wrote/updated tests for new/changed code
- [x] I removed all sensitive data before every commit, including API endpoints and keys
- [x] I ran `fastlane ci` without errors

### CODE REVIEW

File: Example/Controllers/Extensions/ViewController+MiniAppUserInfoProtocol.swift [hunk 1]
```diff
@@ -16,6 +16,14 @@ extension ViewController {
         return userProfilePhoto
     }
 
+    func getContacts() -> [MAContact]? {
+        guard let userProfile = getProfileSettings(), let contactList = userProfile.contactList else {
+            return nil
+        }
+        let contacts = contactList.map { MAContact(id: $0.id) }
```
Comment (ts-leojoseph-r) [inline Example/Controllers/Extensions/ViewController+MiniAppUserInfoProtocol.swift:23]: You can return them just by `return contactList.map { MAContact(id: $0.id) }`

File: Example/Models/UserProfileModel.swift [hunk 2]
```diff
@@ -3,15 +3,23 @@ import UIKit
...
 }
 
+struct Contact: Codable {
+    var id: String
+
+    init(id: String) {
+        self.id = id
+    }
+}
+
```
Comment (ts-leojoseph-r) [inline Example/Models/UserProfileModel.swift:22]: We can use the public class `MAContact` right ? This seems a duplicate just for the sample app

File: MiniApp/Classes/Protocols/MiniAppUserInfoDelegate.swift [hunk 3]
```diff
@@ -37,6 +44,14 @@ public class MATokenInfo: Codable {
...
 }
 
+public class MAContact: Codable {
+    let id: String
+
+    public init(id: String) {
+        self.id = id
+    }
+}
+
```
Comment (ts-leojoseph-r) [inline MiniApp/Classes/Protocols/MiniAppUserInfoDelegate.swift:54]: Some developer documentation maybe ?

**Review by ts-leojoseph-r — COMMENTED** [review]
Thanks for the PR @minh-rakuten . Too bad that I worked on this change already in local was thinking to push it tomorrow 😆 . Never mind. Added few comments, have a look at them.

File: Example/Models/UserProfileModel.swift (hunk 2 above)
Comment (minh-rakuten) [reply Example/Models/UserProfileModel.swift:22]: I don't know why but I cannot import `MAContact`, neither of `MATokenInfo`.

Comment (corycaywood) [reply Example/Models/UserProfileModel.swift:22]: I think this should work with `MAContact` - it was working for me when I tried it. 
@minh-rakuten Did you try cleaning the XCode build folder? Also make sure to add `import MiniApp` to this file.

File: MiniApp/Classes/Protocols/MiniAppUserInfoDelegate.swift [hunk 4]
```diff
@@ -9,6 +9,9 @@ public protocol MiniAppUserInfoDelegate: class {
     /// Interface that is used to retrieve the Image URI
     func getProfilePhoto() -> String?
 
+    /// Interface that is used to retrieve the Contact list
+    func getContacts() -> [MAContact]?
```
Comment (corycaywood) [inline MiniApp/Classes/Protocols/MiniAppUserInfoDelegate.swift:13]: This one also needs to be added to the user guide.

**Review by corycaywood — COMMENTED** [review]
Looking good 👍  

@minh-rakuten is an iOS developer now? 😄

File: Example/Models/UserProfileModel.swift (hunk 2 above)
Comment (minh-rakuten) [reply Example/Models/UserProfileModel.swift]: Thanks. Re-build in XCode works for me.

**Review by ts-leojoseph-r — APPROVED** [review]
Looks good to me :+1: 👏

**Review by corycaywood — APPROVED** [review]
Looks good! 👍

**Pull request merged**